pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/mercury/MercuryOrderDao.java

```diff
@@ -244,47 +295,56 @@ public void insertMercuryOrders(List<MercuryOrderDto> newOrders, String json) {
+     */
+    public List<MercuryOrderDto> getByOrderId(String orderId) {
+        List<MercuryOrderDto> ordersWithOrderId = new ArrayList<>();
+        SimpleResult result = TransactionWrapper.inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult();
+            try {
+                PreparedStatement stmt = conn.prepareStatement(SQL_GET_MERCURY_ORDER_ID);
+                stmt.setString(1, orderId);
+                try {
+                    ResultSet rs = stmt.executeQuery();
                     if (rs.next()) {
-                        execResult.resultValue = rs.getInt(1);
+                        MercuryOrderDto mercurySequencingDto = new MercuryOrderDto.Builder()
+                                .withOrderId(rs.getString(DBConstants.MERCURY_ORDER_ID))
+                                .withMercuryPdoId(rs.getString(DBConstants.MERCURY_PDO_ID))
+                                .withBarcode(rs.getString(DBConstants.MERCURY_BARCODE))
+                                .withDdpInstanceId(rs.getInt(DBConstants.DDP_INSTANCE_ID))
+                                .withOrderDate(rs.getLong(DBConstants.MERCURY_ORDER_DATE))
+                                .withStatusDate(rs.getLong(DBConstants.MERCURY_STATUS_DATE))
+                                .withOrderStatus(rs.getString(DBConstants.MERCURY_ORDER_STATUS))
+                                .withDdpParticipantId(rs.getString(DBConstants.DDP_PARTICIPANT_ID))
+                                .withTissueId(rs.getInt(DBConstants.TISSUE_ID))
+                                .withDsmKitRequestId(rs.getInt(DBConstants.DSM_KIT_REQUEST_ID)).build();
+                        ordersWithOrderId.add(mercurySequencingDto);
                     }
+                } catch (SQLException e) {
+                    dbVals.resultException = new DsmInternalError("Unable to query db for mercury order with order number "
+                            + orderId, e);
                 }
-            } catch (SQLException ex) {
-                execResult.resultException = ex;
+            } catch (SQLException e) {
+                dbVals.resultException = new DsmInternalError("Unable to query db for mercury order with order number "
+                        + orderId, e);
             }
-            return execResult;
-
+            return dbVals;
         });
-        if (results.resultException != null) {
-            throw new DsmInternalError(String.format("Error inserting mercury order for order with json {}", messageJson),
-                    results.resultException);
+        if (result.resultException != null) {
+            throw new DsmInternalError(result.resultException);
         }
-        return (int) results.resultValue;
+        log.info("Found {} barcodes for order number {} ", ordersWithOrderId.size(), orderId);
+        return ordersWithOrderId;
+    }
+
+    public boolean isSequencingOrderValidForPhiReport(@NonNull List<MercuryOrderDto> orders, @NonNull String ddpParticipantId,
```

Comment:
I think I had an earlier comment about this--what "valid" means is unclear, so moving this to MercuryOrderDto.containsParticipantForInstance() and then iterating through the list would help with clarity and code re-use.

Reply:
I misunderstood that comment and instead added what it is valid for (hence the "forPhiReport" part), I changed it now and moved it. Thx!